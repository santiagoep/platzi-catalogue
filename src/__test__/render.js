/* istanbul ignore file */
/* eslint-disable import/no-extraneous-dependencies */
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { makeMockedStore } from '@store/index';
import mockedStore from '@__mocks__/store/index';
import lightTheme from '@assets/styles/themes/light';

const WrapperComponent = ({ children, config, initialState }) => {
  if (config && config.theme && config.redux) {
    const store = makeMockedStore(initialState || mockedStore);
    return (
      <Provider store={store}>
        <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
      </Provider>
    );
  }
  if (config && config.redux) {
    const store = makeMockedStore(initialState || mockedStore);
    return <Provider store={store}>{children}</Provider>;
  }
  if (config && config.theme) {
    return <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>;
  }
  return children;
};

const renderWithProviders = (
  { redux, theme } = {},
  initialState,
  getMockedStore
) => {
  const renderProviders = (Component) => (
    <WrapperComponent
      config={{ redux, theme }}
      initialState={initialState}
      getMockedStore={getMockedStore}
    >
      {Component}
    </WrapperComponent>
  );

  return {
    mount: (Component) => mount(renderProviders(Component)),
    render: (Component) => render(renderProviders(Component)),
    shallow: (Component) => shallow(renderProviders(Component)),
    renderer: (Component) => Renderer.create(renderProviders(Component))
  };
};

export default renderWithProviders;
