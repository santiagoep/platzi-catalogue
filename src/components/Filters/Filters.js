import PropTypes from 'prop-types';

import useFilters from '@hooks/useFilters';
import StyledFilters from './Filters.styled';
import Search from '@components/Search/Search';
import Select from '@components/Select/Select';
import FancyLine from '@components/FancyLine/FancyLine';
import Typography from '@components/Typography/Typography';

const Filters = ({ onFilter }) => {
  const [{ options }, { internalOnFilter }] = useFilters({
    onFilter
  });

  return (
    <StyledFilters data-testid='filters'>
      <Typography type='h6' fontWeight='bold' color='quaternary'>
        Filtros
      </Typography>
      <FancyLine />
      <Search
        id='query-text-filters'
        onSearch={(queryText) => internalOnFilter(queryText, 'queryText')}
      />
      <Select
        id='sort-filters'
        label='Ordenar'
        defaultValue='Ordenar por...'
        options={options}
        onSelect={(value) => internalOnFilter(value, 'sort')}
      />
    </StyledFilters>
  );
};

Filters.propTypes = {
  onFilter: PropTypes.func
};

Filters.defaultProps = {
  onFilter: () => {}
};

export default Filters;
