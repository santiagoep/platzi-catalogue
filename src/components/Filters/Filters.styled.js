import styled from 'styled-components';

export default styled.div`
  padding: ${({ theme }) => theme.utils.spacing(4)};
  .filters__clean {
    cursor: pointer;
  }
`;
