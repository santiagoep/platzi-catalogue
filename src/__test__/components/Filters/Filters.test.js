import render from '../../render';
import Filters from '@components/Filters/Filters';

describe('<Filters />', () => {
  it('Should render snapshot', () => {
    const { mount } = render({ theme: true });
    const component = mount(<Filters />);
    expect(component).toMatchSnapshot();
  });
  it('Should search', () => {
    const { mount } = render({ theme: true });
    const spy = jest.fn();
    const component = mount(<Filters onFilter={spy} />);
    component
      .find('#query-text-filters')
      .last()
      .simulate('change', { target: { value: 'Test' } });
    expect(spy).toHaveBeenCalledWith({
      queryText: 'Test',
      sort: ''
    });
  });
  it('Should sort', () => {
    const { mount } = render({ theme: true });
    const spy = jest.fn();
    const component = mount(<Filters onFilter={spy} />);
    component
      .find('#sort-filters')
      .last()
      .simulate('change', {
        target: { value: 'price_from_lowest_to_highest' }
      });
    expect(spy).toHaveBeenCalledWith({
      queryText: '',
      sort: 'price_from_lowest_to_highest'
    });
  });
  it('Should clean filters', () => {
    const { mount } = render({ theme: true });
    const spy = jest.fn();
    const component = mount(<Filters onFilter={spy} />);
    component
      .find('#query-text-filters')
      .last()
      .simulate('change', { target: { value: 'Test' } });
    component.find('.filters__clean').last().simulate('click');
    expect(spy).toHaveBeenCalledWith({
      queryText: '',
      sort: ''
    });
  });
});
