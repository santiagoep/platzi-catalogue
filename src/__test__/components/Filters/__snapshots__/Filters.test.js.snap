// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> Should render snapshot 1`] = `
<WrapperComponent
  config={
    Object {
      "redux": undefined,
      "theme": true,
    }
  }
>
  <Component
    theme={
      Object {
        "borderRadius": Object {
          "button": "100px",
          "large": "18px",
          "medium": "8px",
          "none": "unset",
          "small": "3px",
          "total": "50%",
        },
        "boxShadows": Object {
          "main": "1px 2px 4px 0 rgba(21,99,157,.16)",
          "secondary": "0 0 0 1px #0791e6",
        },
        "breakpoints": Object {
          "desktop": "@media only screen and (min-width: 1280px)",
          "laptop": "@media only screen and (min-width: 1024px)",
          "mobile": "@media only screen and (min-width: 320px)",
          "screen": "@media only screen and (min-width: 1920px)",
          "smallMobile": "@media only screen and (min-width: 0px)",
          "tablet": "@media only screen and (min-width: 640px)",
        },
        "colors": Object {
          "bg": Object {
            "primary": "linear-gradient(90deg,#1c3643,#273b47 25%,#1e5372)",
            "quaternary": "#F3F8FB",
            "secondary": "white",
            "tertiary": "#0791e6",
          },
          "border": Object {
            "primary": "#A8B1C0",
            "secondary": "white",
          },
          "font": Object {
            "primary": "#131D30",
            "quaternary": "#A8B1C0",
            "secondary": "#98ca3f",
            "tertiary": "white",
          },
          "separator": Object {
            "primary": "#A8B1C0",
            "secondary": "#0791e6",
          },
        },
        "containerWidth": Object {
          "desktop": Object {
            "maxWidth": "85%",
            "width": "85%",
          },
          "fluid": Object {
            "maxWidth": "100%",
            "width": "100%",
          },
          "limitFluid": Object {
            "maxWidth": "1920px",
            "width": "100%",
          },
          "mobile": Object {
            "maxWidth": "92%",
            "width": "92%",
          },
          "screen": Object {
            "maxWidth": "75%",
            "width": "75%",
          },
          "smallMobile": Object {
            "maxWidth": "92%",
            "width": "92%",
          },
          "tablet": Object {
            "maxWidth": "85%",
            "width": "85%",
          },
        },
        "font": Object {
          "family": Object {
            "primary": "'Lato', sans-serif",
          },
          "letterSpacing": Object {
            "medium": "0.5rem",
            "normal": "0rem",
          },
          "lineHeight": Object {
            "medium": "0.6rem",
            "normal": "normal",
          },
          "size": Object {
            "h1": "3.75rem",
            "h2": "3rem",
            "h3": "2.5rem",
            "h4": "2.25rem",
            "h5": "1.625rem",
            "h6": "1.5rem",
            "help": "0.625em",
            "paragraph1": "1.125rem",
            "paragraph2": "1rem",
            "paragraph3": "0.875rem",
            "paragraph4": "0.75rem",
            "small": "0.5rem",
            "subtitle1": "1.1375rem",
            "subtitle2": "1.25rem",
            "subtitle3": "1.375rem",
          },
        },
        "name": "LIGHT",
        "transitions": Object {
          "button": "background-color 0.4s ease, color 0.4s ease, border-color 0.4s ease",
        },
        "utils": Object {
          "spacing": [Function],
        },
        "zIndex": Object {
          "backward": -1,
          "fixed": 2,
          "forward": 1,
          "higher": 3,
          "initial": 0,
          "super": 100,
        },
      }
    }
  >
    <Filters
      onFilter={[Function]}
    >
      <Filtersstyled
        data-testid="filters"
      >
        <div
          className="Filtersstyled-sc-8y7woq-0 iTDjCP"
          data-testid="filters"
        >
          <Typography
            className="typography"
            color="quaternary"
            fontFamily="primary"
            fontStyle="normal"
            fontWeight="bold"
            letterSpacing="normal"
            lineHeight="normal"
            textDecoration="none"
            textTransform="none"
            type="h6"
          >
            <Typographystyled
              as="h6"
              className="typography"
              color="quaternary"
              fontFamily="primary"
              fontStyle="normal"
              fontWeight="bold"
              letterSpacing="normal"
              lineHeight="normal"
              textDecoration="none"
              textTransform="none"
              type="h6"
            >
              <h6
                className="Typographystyled-sc-1pbtdtz-0 esWlpf typography"
                color="quaternary"
                fontFamily="primary"
                fontStyle="normal"
                fontWeight="bold"
                letterSpacing="normal"
                textDecoration="none"
                type="h6"
              >
                Filtros
              </h6>
            </Typographystyled>
          </Typography>
          <FancyLine
            height="1px"
            lineColor="secondary"
          >
            <FancyLinestyled
              height="1px"
              lineColor="secondary"
            >
              <div
                className="FancyLinestyled-sc-1ffx9pt-0 gYjXDl"
                height="1px"
              />
            </FancyLinestyled>
          </FancyLine>
          <Search
            id="query-text-filters"
            onSearch={[Function]}
            placeholder="Buscar..."
          >
            <Searchstyled__StyledContainer>
              <div
                className="Searchstyled__StyledContainer-nj7k97-0 fmXmHm"
              >
                <Searchstyled__StyledSearch
                  id="query-text-filters"
                  onChange={[Function]}
                  placeholder="Buscar..."
                  value=""
                >
                  <input
                    className="Searchstyled__StyledSearch-nj7k97-1 ZPjyu"
                    id="query-text-filters"
                    onChange={[Function]}
                    placeholder="Buscar..."
                    value=""
                  />
                </Searchstyled__StyledSearch>
                <Searchstyled__StyledIcon>
                  <div
                    className="Searchstyled__StyledIcon-nj7k97-2 jBicbR"
                  >
                    <Image
                      alt="Search Icon"
                      height={18}
                      src="/icons/search.svg"
                      width={18}
                    >
                      <div
                        style={
                          Object {
                            "boxSizing": "border-box",
                            "display": "inline-block",
                            "margin": 0,
                            "maxWidth": "100%",
                            "overflow": "hidden",
                            "position": "relative",
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "boxSizing": "border-box",
                              "display": "block",
                              "maxWidth": "100%",
                            }
                          }
                        >
                          <img
                            alt=""
                            aria-hidden={true}
                            role="presentation"
                            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgiIGhlaWdodD0iMTgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                            style={
                              Object {
                                "display": "block",
                                "maxWidth": "100%",
                              }
                            }
                          />
                        </div>
                        <img
                          alt="Search Icon"
                          decoding="async"
                          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                          style={
                            Object {
                              "border": "none",
                              "bottom": 0,
                              "boxSizing": "border-box",
                              "display": "block",
                              "height": 0,
                              "left": 0,
                              "margin": "auto",
                              "maxHeight": "100%",
                              "maxWidth": "100%",
                              "minHeight": "100%",
                              "minWidth": "100%",
                              "objectFit": undefined,
                              "objectPosition": undefined,
                              "padding": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "visibility": "hidden",
                              "width": 0,
                            }
                          }
                        />
                      </div>
                    </Image>
                  </div>
                </Searchstyled__StyledIcon>
              </div>
            </Searchstyled__StyledContainer>
          </Search>
          <Select
            defaultValue="Ordenar por..."
            id="sort-filters"
            label="Ordenar"
            onSelect={[Function]}
            options={
              Array [
                Object {
                  "display": "Precio de mayor a menor",
                  "value": "price_from_highest_to_lowest",
                },
                Object {
                  "display": "Precio de menor a mayor",
                  "value": "price_from_lowest_to_highest",
                },
              ]
            }
          >
            <Typography
              as="label"
              className="typography"
              color="primary"
              fontFamily="primary"
              fontStyle="normal"
              fontWeight="normal"
              htmlFor="sort-filters"
              letterSpacing="normal"
              lineHeight="normal"
              textDecoration="none"
              textTransform="none"
              type="paragraph4"
            >
              <Typographystyled
                as="label"
                className="typography"
                color="primary"
                fontFamily="primary"
                fontStyle="normal"
                fontWeight="normal"
                htmlFor="sort-filters"
                letterSpacing="normal"
                lineHeight="normal"
                textDecoration="none"
                textTransform="none"
                type="paragraph4"
              >
                <label
                  className="Typographystyled-sc-1pbtdtz-0 bEYlVL typography"
                  color="primary"
                  fontFamily="primary"
                  fontStyle="normal"
                  fontWeight="normal"
                  htmlFor="sort-filters"
                  letterSpacing="normal"
                  textDecoration="none"
                  type="paragraph4"
                >
                  Ordenar
                </label>
              </Typographystyled>
            </Typography>
            <Selectstyled__StyledContainer>
              <div
                className="Selectstyled__StyledContainer-sc-1d2yf14-0 jcUCjC"
              >
                <Selectstyled__StyledSelect
                  defaultValue=""
                  id="sort-filters"
                  onChange={[Function]}
                >
                  <select
                    className="Selectstyled__StyledSelect-sc-1d2yf14-1 dCxTJM"
                    defaultValue=""
                    id="sort-filters"
                    onChange={[Function]}
                  >
                    <Selectstyled__StyledOption
                      value=""
                    >
                      <option
                        className="Selectstyled__StyledOption-sc-1d2yf14-2"
                        value=""
                      >
                        Ordenar por...
                      </option>
                    </Selectstyled__StyledOption>
                    <Selectstyled__StyledOption
                      key="select__option-price_from_highest_to_lowest-0"
                      value="price_from_highest_to_lowest"
                    >
                      <option
                        className="Selectstyled__StyledOption-sc-1d2yf14-2"
                        value="price_from_highest_to_lowest"
                      >
                        Precio de mayor a menor
                      </option>
                    </Selectstyled__StyledOption>
                    <Selectstyled__StyledOption
                      key="select__option-price_from_lowest_to_highest-1"
                      value="price_from_lowest_to_highest"
                    >
                      <option
                        className="Selectstyled__StyledOption-sc-1d2yf14-2"
                        value="price_from_lowest_to_highest"
                      >
                        Precio de menor a mayor
                      </option>
                    </Selectstyled__StyledOption>
                  </select>
                </Selectstyled__StyledSelect>
                <Selectstyled__StyledIcon>
                  <div
                    className="Selectstyled__StyledIcon-sc-1d2yf14-3 TCQxc"
                  >
                    <Image
                      alt="Arrow Icon"
                      height={18}
                      src="/icons/bottom-arrow.svg"
                      width={18}
                    >
                      <div
                        style={
                          Object {
                            "boxSizing": "border-box",
                            "display": "inline-block",
                            "margin": 0,
                            "maxWidth": "100%",
                            "overflow": "hidden",
                            "position": "relative",
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "boxSizing": "border-box",
                              "display": "block",
                              "maxWidth": "100%",
                            }
                          }
                        >
                          <img
                            alt=""
                            aria-hidden={true}
                            role="presentation"
                            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgiIGhlaWdodD0iMTgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                            style={
                              Object {
                                "display": "block",
                                "maxWidth": "100%",
                              }
                            }
                          />
                        </div>
                        <img
                          alt="Arrow Icon"
                          decoding="async"
                          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                          style={
                            Object {
                              "border": "none",
                              "bottom": 0,
                              "boxSizing": "border-box",
                              "display": "block",
                              "height": 0,
                              "left": 0,
                              "margin": "auto",
                              "maxHeight": "100%",
                              "maxWidth": "100%",
                              "minHeight": "100%",
                              "minWidth": "100%",
                              "objectFit": undefined,
                              "objectPosition": undefined,
                              "padding": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "visibility": "hidden",
                              "width": 0,
                            }
                          }
                        />
                      </div>
                    </Image>
                  </div>
                </Selectstyled__StyledIcon>
              </div>
            </Selectstyled__StyledContainer>
          </Select>
        </div>
      </Filtersstyled>
    </Filters>
  </Component>
</WrapperComponent>
`;
